.pre-commit-config.yaml

```diff
@@ -5,14 +5,15 @@ repos:
     -   id: check-yaml
     -   id: end-of-file-fixer
     -   id: trailing-whitespace
+-   repo: https://github.com/psf/black
+    rev: 21.6b0
+    hooks:
+    -   id: black
 -   repo: https://gitlab.com/pycqa/flake8
-    rev: 3.7.8
+    rev: 3.9.2
     hooks:
     -   id: flake8
         types: [file, python]
-        # only check for unused imports for now, as long as
-        # the code is not fully PEP8 compatible
-        args: [--select=F401]
```

Comment:
This might be a problem when working on the examples which allow delayed import statements when using the notebook style code layout and the non-blackified files such as benchmark scripts, no?

Reply:
https://github.com/scikit-learn/scikit-learn/blob/main/setup.cfg#L54-L57

Reply:
Hmm, because we don't want to use `# noqa` in examples? I think we should come up with a config that would make `flake8 .` just work;  related https://github.com/scikit-learn/scikit-learn/issues/20296 .

But since this requires more discussion, let me revert the flake8 change, here having black setup would be already good.

Reply:
OK, then with `per-file-ignores` it should be fine.

Reply:
I approved the PR, assuming you tested it and it does not yield spurious rejections when working on the examples.